pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -298,13 +316,23 @@ func (h *AutoscalersController) syncAutoscalers(key interface{}) error {
 			return ErrIsEmpty
 		}
 		new := h.hpaProc.ProcessHPAs(hpa)
+		if len(new)+h.metricsProcessedCount > maxMetricsCount {
+			log.Warnf("Currently processing %d metrics, skipping %s/%s as we can't process more than %d metrics",
```

Comment:
also can we raise this warning in the status command output?

Reply:
Yes, that's what I wanted to do. It requires making the `metricsProcessedCount` public and doing some changes in the status package (not too much I think). 
I figured it was more important to first validate the logic and then discuss the need to have this more visible